Sources/Client/WebSocket/WebSocket.swift

```diff
@@ -31,33 +29,31 @@ final class WebSocket {
     }
     
     private lazy var handshakeTimer =
-        RepeatingTimer(timeInterval: .seconds(WebSocket.pingTimeInterval), queue: webSocket.callbackQueue) { [weak self] in
+        RepeatingTimer(timeInterval: .seconds(WebSocket.pingTimeInterval), queue: provider.callbackQueue) { [weak self] in
             self?.logger?.log("🏓➡️", level: .info)
-            self?.webSocket.write(ping: .empty)
+            self?.provider.sendPing()
     }
     
     /// Checks if the web socket is connected and `connectionId` is not nil.
-    var isConnected: Bool { connectionId != nil && webSocket.isConnected }
+    var isConnected: Bool { connectionId != nil && provider.isConnected }
     
-    init(_ urlRequest: URLRequest,
-         stayConnectedInBackground: Bool = true,
+    init(_ provider: WebSocketProvider,
+         options: WebSocketOptions,
          logger: ClientLogger? = nil,
          onEvent: @escaping (Event) -> Void = { _ in }) {
-        self.stayConnectedInBackground = stayConnectedInBackground
+        self.provider = provider
+        self.options = options
         self.logger = logger
         self.onEvent = onEvent
-        webSocket = Starscream.WebSocket(request: urlRequest)
-        webSocket.callbackQueue = DispatchQueue(label: "io.getstream.Chat.WebSocket", qos: .userInitiated)
-        webSocketInitiated = true
-        webSocket.delegate = self
+        self.provider.delegate = self
     }
     
     init() {
-        webSocket = .init(url: BaseURL.placeholderURL)
-        webSocketInitiated = false
-        stayConnectedInBackground = false
+        provider = EmptyWebSocketProvider(request: URLRequest(url: BaseURL.placeholderURL), callbackQueue: .main)
+        options = []
         logger = nil
         onEvent = { _ in }
+        provider.delegate = self
     }
```

Comment:
I think this whole `init` is redundant but it's probably a topic for another PR.

Reply:
Yes, I don't like it too. I'll think about that.